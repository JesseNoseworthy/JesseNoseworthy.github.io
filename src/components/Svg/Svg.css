@import '../../styles/variables.css';
@import '../../styles/mixins.css';

:local(.Svg) {
  @mixin FlexCenter;
  width: 100%;
}