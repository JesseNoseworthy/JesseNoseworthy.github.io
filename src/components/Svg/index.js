import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Svg.css';

class Svg extends Component {
  parseSource(src) {
    return { __html: src };
  }

  render() {
    const {
      src,
      className,
      ...others
    } = this.props;

    return (
      <div
        className={classNames(styles.Svg, className)}
        dangerouslySetInnerHTML={this.parseSource(src)}
        {...others}
      />
    );
  }
}

Svg.propTypes = {
  src: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Svg;