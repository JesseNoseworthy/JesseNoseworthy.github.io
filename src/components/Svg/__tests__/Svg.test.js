import React from 'react';
import renderer from 'react-test-renderer';
import Svg from '../index';

describe('Svg', () => {
  describe('Snapshot tests', () => {
    it('default state (empty)', () => {
      const component = renderer.create(
        <Svg src={'/icon.svg'} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('populated attributes', () => {
      const component = renderer.create(
        <Svg
          src={'/icon.svg'}
          className={'my-class'}
        />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
