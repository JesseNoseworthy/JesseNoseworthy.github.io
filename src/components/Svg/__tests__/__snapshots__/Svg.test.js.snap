exports[`Svg Snapshot tests default state (empty) 1`] = `
<div
  className=""
  dangerouslySetInnerHTML={
    Object {
      "__html": "/icon.svg",
    }
  } />
`;

exports[`Svg Snapshot tests populated attributes 1`] = `
<div
  className="my-class"
  dangerouslySetInnerHTML={
    Object {
      "__html": "/icon.svg",
    }
  } />
`;
